### PR TITLE
Avoid recomputing all_assets_map on every change

### PR DESCRIPTION
**Pull request by alexkirsz**

### CODE REVIEW

**Review by sokra — APPROVED** [review]

File: crates/turbopack-ecmascript/src/chunk/source_map.rs [hunk 1]
```diff
@@ -131,7 +135,7 @@ impl AssetReference for EcmascriptChunkSourceMapAssetReference {
                     EcmascriptChunkEntrySourceMapAsset {
                         chunk_path: path,
                         code: item.code_vc,
-                        hash: item.hash,
+                        id: item.chunk_item.id(),
```
Comment (jridgewell) [inline crates/turbopack-ecmascript/src/chunk/source_map.rs:138]: If the `item.hash` is already computed, why can't we just use it instead?

Will that force a new `EcmascriptChunkEntrySourceMapAssetVc` every time the hash changes?

**Review by jridgewell — CHANGES_REQUESTED** [review]
This'll break https://github.com/vercel/turbo/blob/b3fad7d1ce2514953ebed198b2217a2c833dc976/crates/next-core/src/nodejs/mod.rs#L407-L416 and https://github.com/vercel/turbo/blob/b3fad7d1ce2514953ebed198b2217a2c833dc976/crates/next-core/src/source_map/content_source.rs#L92-L95 until https://github.com/vercel/turbo/pull/2426 is merged.

File: crates/turbopack-ecmascript/src/chunk/source_map.rs (hunk 1 above)
Comment (sokra) [reply crates/turbopack-ecmascript/src/chunk/source_map.rs:138]: See followup PR.

**Review by jridgewell — APPROVED** [review]

**Comment by jaredpalmer** [comment]
We merging?

**Pull request merged**